client/src/interface/components/ChangePermission/index.js

```diff
@@ -1,192 +1,203 @@
+    handleModeInputChange = (ev) => {
+        const value = ev.target.value
+        if (value.length > 3) {
+            return
+        }
+        if (value.length !== 0 && !/^[0-7]+$/.test(value)) {
+            return
+        }
+        this.setState({
+            modeInput: value,
+            modeCheck: parseInt(value.padEnd(3, '0'), 8)
+        })
+    }
+    handleModeInputBlur = (_) => {
+        if (this.state.modeInput.length !== 3) {
+            this.setState({
+                modeCheck: (this.state.mode & 0o777),
+                modeInput: (this.state.mode & 0o777).toString(8).padStart(3, '0')
+            })
+        } else {
+            this.setState({
+                mode: this.state.mode & 0xfe00 | this.state.modeCheck
+            })
+        }
+    }
+    update_id_mode = (_cwd, _name, _mode) => {
+        this.cwd = _cwd
+        this.setState({
+            name: _name,
+            mode: _mode,
+            modeCheck: (_mode & 0o777),
+            modeInput: (_mode & 0o777).toString(8).padStart(3, '0')
+        })
     }
 
     render() {
         const {open} = this.props;
+        if (!open) {
+            return null
+        }
+
+        const {name, mode, modeCheck, modeInput, applyRecursively} = this.state
+        const entryIsDir = isDir(mode)
+
+        const rows_list = []
+        for (let i = 0; i < 3; i++) {
+            const who = ['Owner', 'Group', 'Others'][i]
+            const row = [<TableCell key={who}><Box fontWeight={'bold'}>{who}</Box></TableCell>]
+
+            for (let j = 0; j < 3; j++) {
+                const op = ['read', 'write', 'execute'][j]
+                const index = i * 3 + j
+                const bit = PERMISSION_BITS[index]
+                row.push(<TableCell key={String(bit)} align={'center'}>
+                    <Tooltip
+                        placement={'top-start'}
+                        title={`If set, the ${who} can ${op} on the ${entryIsDir ? 'directory' : 'file'}`}>
+                        <Checkbox
+                            name={String(bit)}
+                            onChange={this.handleModeChange}
+                            checked={Boolean(modeCheck & bit)}/>
+                    </Tooltip>
+                </TableCell>)
+            }
+            rows_list.push(
+                <TableRow key={`row_${i}`}>
+                    {row}
+                </TableRow>
+            )
+        }
+
         return (
             <Dialog
                 open={open}
-                onClose={this.handleClose}
                 fullWidth={true}
                 maxWidth={'xs'}
                 aria-labelledby="change permission"
             >
-                <DialogTitle>Change Permission</DialogTitle>
-                <DialogTitle>{this.state.id}</DialogTitle>
+                <DialogTitle>{name} - Permissions</DialogTitle>
                 <div className={'change-permission-content-wrapper'}>
-                    <FormGroup row>
```

Comment:
see above for the reasonings for refactoring this

let me know if you dislike this approach and we can discuss that